### PR TITLE
RI-8220: Add Aggregate tab to Array key details

### PR DESCRIPTION
**Pull request by pawelangelow**
# What

Adds the **Aggregate** tab to the Array key details view, exposing the `POST /array/aggregate` (`AROP`) endpoint to the UI. Users can pick a range, an operation, and run aggregations from the key view.

**Form:**
- `start` / `end` index inputs (u64-safe strings, client-side validation against the 1 000 000-element span cap)
- Operation select: `SUM`, `MIN`, `MAX`, `AND`, `OR`, `XOR`, `MATCH`, `USED`
- `value` field is shown and required only when `MATCH` is selected
- **Run** / **Reset** controls; result is copyable

# Screenshots

| Dark    | Light |
| -------- | ------- |
| Screenshot 2026-06-18 at 12 54 24 | Screenshot 2026-06-18 at 12 53 39  |
| Screenshot 2026-06-18 at 12 54 55 | Screenshot 2026-06-18 at 12 54 45     |

---

> [!NOTE]
> **Low Risk**
> Browser-only feature on a new aggregate code path, isolated from range/scan via separate abort controller and slice; extensive unit tests cover validation, refresh replay, and edge cases.
> 
> **Overview**
> Replaces the Array key **Aggregate** tab placeholder with a full **AROP** (`POST array/aggregate`) workflow: range + operation form, result panel, and Redux wiring separate from the View tab’s range/scan state.
> 
> **UI:** `ArrayAggregateForm` mirrors the View tab’s range form (index validation, 1M span cap, optional MATCH value, command preview, Run/Reset). `AggregateTab` shows loading/errors, displays results as raw strings (BigInt-safe) or `(nil)`, and hides stale state while the selected key is not ready.
> 
> **State:** New `browser.array.aggregate` slice + `aggregateArray` thunk with its own `AbortController` (so View range/scan does not cancel AROP). `refreshArray` replays the last aggregate when `hasResult` is set, using `resetData: false` to avoid loader flash on header refresh.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ec32d36f280e22802b4260ec14ad464ef4a89271. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/array-aggregate-form/ArrayAggregateForm.constants.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export const RUN_BUTTON_LABEL = 'Run'
+export const RESET_TOOLTIP = 'Reset to defaults'
+export const INVALID_INDEX_MESSAGE =
+  'Index must be a valid 64-bit unsigned integer'
+
+/**
+ * Mirror of the backend's `ARRAY_RANGE_MAX_ELEMENTS` cap — AROP reuses
+ * `assertValidRange` so the same span limit applies.
+ */
+export const ARRAY_RANGE_MAX_SPAN = 1_000_000n
```
Comment (VaskoAtanasovRedis) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/array-aggregate-form/ArrayAggregateForm.constants.ts:14]: nit: comments says it mirrors `ARRAY_RANGE_MAX_ELEMENTS` ; should we also call it `ARRAY_RANGE_MAX_ELEMENTS` here?

File: redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/aggregate-tab/AggregateTab.tsx [hunk 2]
```diff
@@ -1,13 +1,114 @@
...
-  </S.TabPlaceholder>
-)
+export interface AggregateTabProps {
+  keyProp: RedisResponseBuffer | null
+}
+
+const AGGREGATE_TAB_TEST_ID = 'array-aggregate-tab'
+const NIL_RESULT_LABEL = '(nil)'
+
+const AggregateTab = ({ keyProp }: AggregateTabProps) => {
```
Comment (VaskoAtanasovRedis) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/aggregate-tab/AggregateTab.tsx:18]: I think these belong in `AggregateTab.types.ts`

File: redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/aggregate-tab/AggregateTab.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+const keyBuffer = stringToBuffer('numbers')
+
+describe('AggregateTab', () => {
+  beforeEach(() => {
+    mockUseArrayAggregateQuery.mockReset()
+    mockUseArrayAggregateQuery.mockReturnValue({ ...baseHookResult })
+  })
+
+  it('renders the form and an empty results area by default', () => {
+    render(<AggregateTab keyProp={keyBuffer} />)
```
Comment (VaskoAtanasovRedis) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/aggregate-tab/AggregateTab.spec.tsx]: I think similar tests centralize these via a `renderComponent(propsOverride?)`

File: redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/array-aggregate-form/ArrayAggregateForm.tsx [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  ARRAY_AGGREGATE_FORM_TEST_ID as TEST_ID,
+  ARRAY_RANGE_MAX_SPAN,
+  INVALID_INDEX_MESSAGE,
+  INVALID_RANGE_TOO_LARGE_MESSAGE,
+  OPERATION_OPTIONS,
+  RESET_TOOLTIP,
+  RUN_BUTTON_LABEL,
+} from './ArrayAggregateForm.constants'
+import { ArrayAggregateFormProps } from './ArrayAggregateForm.types'
+
```
Comment (VaskoAtanasovRedis) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/array-aggregate-form/ArrayAggregateForm.tsx:29]: I think this whole form is lacking unit tests

File: redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/array-aggregate-form/ArrayAggregateForm.constants.ts (hunk 1 above)
Comment (pawelangelow) [reply redisinsight/ui/src/pages/browser/modules/key-details/components/array-details/array-aggregate-form/ArrayAggregateForm.constants.ts:14]: Keeping `ARRAY_RANGE_MAX_SPAN` here for consistency with the sibling ArrayRangeForm.constants.ts, which already uses that name. The "span" naming is deliberate - it's the BigInt end − start + 1 span the form computes client-side, distinct from the backend's element count it's compared against (hence the comment pointing at `ARRAY_RANGE_MAX_ELEMENTS`). Happy to rename both in a follow-up if you'd prefer the names match the backend exactly.

**Review by VaskoAtanasovRedis — APPROVED** [review]
